### PR TITLE
Feature/support calendar events

### PR DESCRIPTION
**Pull request by baumannsven**

### CODE REVIEW

**Comment by baumannsven** [comment]
@BugBuster1701 Do have seen this?

**Comment by BugBuster1701** [comment]
Not until today. I was ill and will look at it in the next few days. Thank you.

File: src/Resources/contao/classes/ModuleVisitorStatEventsCounter.php [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                return $TemplatePartial->parse();
+            }
+
+            return $arrEventsStatCount;
+        }
+
+        return false;
+    }
+
+    public function generateNewsVisitHitDays($VisitorsID, $limit = 10, $parse = true, $days=7)
```
Comment (BugBuster1701) [inline src/Resources/contao/classes/ModuleVisitorStatEventsCounter.php:164]: `generateEventsVisitHitDays` sollte das doch heißen, oder?

File: src/Resources/contao/modules/ModuleVisitorStat.php [hunk 2]
```diff
@@ -249,6 +250,10 @@ protected function compile()
 				// Isotope
 				$arrVisitorsIsotopeVisitHits[$intAnzCounter]       = ModuleVisitorStatIsotopeProductCounter::getInstance()->generateIsotopeVisitHitTop($objVisitorsID, 20, true);
 
+                // Events
+                $arrVisitorsEventsVisitHits[$intAnzCounter]          = ModuleVisitorStatEventsCounter::getInstance()->generateEventsVisitHitTop($objVisitorsID, 20, true);
+                $arrVisitorsEventsVisitHitsDays[$intAnzCounter]      = ModuleVisitorStatEventsCounter::getInstance()->generateNewsVisitHitDays($objVisitorsID, 20, true, 7);
```
Comment (BugBuster1701) [inline src/Resources/contao/modules/ModuleVisitorStat.php:255]: `generateEventsVisitHitDays`

File: src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithitdays.html5 [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<div class="mod_visitors_be_stat_partial_eventsvisithitdays">
+<!-- NewsHitDays::start -->
```
Comment (BugBuster1701) [inline src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithitdays.html5:2]: EventsHitDays

File: src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithitdays.html5 [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+			<td style="padding-left: 2px; text-align: left; word-break: break-all; word-wrap: break-word;" class="tl_file_list"><?php echo (isset($EventsVisitHitDaysRow['title']) ? $EventsVisitHitDaysRow['title']  : ''); ?></td>
+			<td style="padding-left: 2px; text-align: left; word-break: break-all; word-wrap: break-word;" class="tl_file_list"><?php echo (isset($EventsVisitHitDaysRow['alias']) ? $EventsVisitHitDaysRow['alias'] . ' (' . (isset($EventsVisitHitDaysRow['lang']) ? $EventsVisitHitDaysRow['lang'] . ')' : '') : ''); ?></td>
+			<td style="padding-left: 2px; text-align: right;"                                              class="tl_file_list"><?php echo (isset($EventsVisitHitDaysRow['hits'])  ? $EventsVisitHitDaysRow['visits'] : ''); ?></td>
+			<td style="padding-left: 2px; text-align: right;"                                              class="tl_file_list"><?php echo (isset($EventsVisitHitDaysRow['hits'])  ? $EventsVisitHitDaysRow['hits']   : ''); ?></td>
+		</tr>
+	<?php endforeach; ?>
+<?php endif; ?>
+	</tbody>
+</table>
+<!-- NewsHitDays::stop -->
```
Comment (BugBuster1701) [inline src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithitdays.html5:26]: EventsHitDays

File: src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithittop.html5 [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<div class="mod_visitors_be_stat_partial_eventsvisithittop">
+<!-- NewsHitTop::start -->
```
Comment (BugBuster1701) [inline src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithittop.html5:2]: EventsHitTop

File: src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithittop.html5 [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+			    <td style="padding-left: 2px; text-align: left; word-break: break-all; word-wrap: break-word;" class="tl_file_list"><?php echo (isset($EventsVisitHitTopRow['title']) ? $EventsVisitHitTopRow['title']  : ''); ?></td>
+			    <td style="padding-left: 2px; text-align: left; word-break: break-all; word-wrap: break-word;" class="tl_file_list"><?php echo (isset($EventsVisitHitTopRow['alias']) ? $EventsVisitHitTopRow['alias'] . ' (' . (isset($EventsVisitHitTopRow['lang']) ? $EventsVisitHitTopRow['lang'] . ')' : '') : ''); ?></td>
+			    <td style="padding-left: 2px; text-align: right;"                                              class="tl_file_list"><?php echo (isset($EventsVisitHitTopRow['hits'])  ? $EventsVisitHitTopRow['visits'] : ''); ?></td>
+			    <td style="padding-left: 2px; text-align: right;"                                              class="tl_file_list"><?php echo (isset($EventsVisitHitTopRow['hits'])  ? $EventsVisitHitTopRow['hits']   : ''); ?></td>
+			</tr>
+	<?php endforeach; ?>
+<?php endif; ?>
+		</tbody>
+</table>
+<!-- NewsHitTop::stop -->
```
Comment (BugBuster1701) [inline src/Resources/contao/templates/mod_visitors_be_stat_partial_eventsvisithittop.html5:26]: EventsHitTop

File: src/Resources/contao/classes/ModuleVisitorStatEventsCounter.php (hunk 1 above)
Comment (BugBuster1701) [inline src/Resources/contao/classes/ModuleVisitorStatEventsCounter.php:164]: generateEventsVisitHitDays

**Review by BugBuster1701 — CHANGES_REQUESTED** [review]
Einige News müssten noch zu Events umbenannt werden

**Pull request merged**